### PR TITLE
[kubeadm] fix kubelet config bug for kubeadm init phase

### LINKED ISSUE #1493: Can not set --cert-dir on init phase kubelet-start
## Is this a BUG REPORT or FEATURE REQUEST?

 BUG REPORT

## Versions

**kubeadm version** (use `kubeadm version`):
```
[root@nixos:/etc]# kubeadm version
kubeadm version: &version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.5", GitCommit:"2166946f41b36dea2c4626f90a77706f426cdea2", GitTreeState:"archive", BuildDate:"1970-01-01T00:00:01Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}

```
**Environment**:
- **Kubernetes version** (use `kubectl version`):
```
[root@nixos:/etc]# kubectl version
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.5", GitCommit:"2166946f41b36dea2c4626f90a77706f426cdea2", GitTreeState:"archive", BuildDate:"1970-01-01T00:00:01Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
The connection to the server localhost:8080 was refused - did you specify the right host or port?


```
- **Cloud provider or hardware configuration**: bare metal
- **OS** (e.g. from /etc/os-release): NixOS
- **Kernel** (e.g. `uname -a`): 4.14

## What happened?

`kubeadm init` supports an option `--cert-dir` 
```
      --cert-dir string                      The path where to save and store the certificates. (default "/etc/kubernetes/pki")
```

When I run `kubeadm init --cert-dir /run/secrets/pki`  it generates the certs succesfully, but `kubelet` fails to start because `config.yaml` contains `/etc/kubernetes/pki/ca.crt` 

```
Apr 07 21:50:24 nixos kubelet[5910]: F0407 21:50:24.301424    5910 server.go:244] unable to load client CA file /etc/kubernetes/pki/ca.crt: open /etc/kubernetes/pki/ca.crt: no such file or directory
Apr 07 21:50:24 nixos systemd[1]: kubelet.service: Main process exited, code=exited, status=255/n/a
Apr 07 21:50:24 nixos systemd[1]: kubelet.service: Failed with result 'exit-code'.

```

## What you expected to happen?

I expected kubelet to pick up the new certs locaiton

## How to reproduce it (as minimally and precisely as possible)?
```
kubeadm init --cert-dir /run/secrets/pki
```

### PR DESCRIPTION
**Pull request by SataQiu**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:
[kubeadm] fix kubelet config bug for kubeadm init phase.

**Which issue(s) this PR fixes**:

Fixes kubernetes/kubeadm#1493

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
thanks for the PR, tentative LGTM.

i think we should instead set the value in the KubeletConfiguration (component configuration).
the rest of the flags for settings are already supported in the KubeletConfiguration and should be moved to there as well.

/assign @fabriziopandini 
@kubernetes/sig-cluster-lifecycle-pr-reviews

**Comment by SataQiu** [comment]
/assign @timothysc 
Could you have a look at this PR?

**Comment by timothysc** [comment]
I'm going to defer to @fabriziopandini on this one, I know he was looking more broadly on a number of certs issues.

**Review by rosti — CHANGES_REQUESTED** [review]
Thanks for tackling this @SataQiu !

We set that value via the component config:
https://github.com/kubernetes/kubernetes/blob/f0a4672e6db7196312ef85731c19eadcbee1da9d/cmd/kubeadm/app/componentconfigs/defaults.go#L91

What we should be doing is to ensure, that we use `internalcfg.CertificatesDir` to get the path (rather than the const).

/hold

**Comment by SataQiu** [comment]
@rosti OK. Thank you for your explanation. I will try further.

**Comment by SataQiu** [comment]
/assign @fabriziopandini 
Could you have a look at this PR?

**Comment by SataQiu** [comment]
@rosti

**Comment by SataQiu** [comment]
@rosti 
I've updated the code.
After a simple test, it worked.

File: cmd/kubeadm/app/componentconfigs/defaults.go
Comment (rosti) [inline cmd/kubeadm/app/componentconfigs/defaults.go]: Looks way better than before, however let's use `filepath.Join` for the actual path construction.

**Review by rosti — COMMENTED** [review]
Thanks @SataQiu !

File: cmd/kubeadm/app/componentconfigs/defaults.go
Comment (SataQiu) [reply cmd/kubeadm/app/componentconfigs/defaults.go]: Thank you so much! 
The code has been updated.

**Review by rosti — APPROVED** [review]
Thanks @SataQiu !
/approve

**Comment by SataQiu** [comment]
/cc @neolit123 @timothysc 
Can you have a look at this PR?

**Review by yagonobre — COMMENTED** [review]
Thanks @SataQiu 
/lgtm

**Pull request merged**
